RecoVertex/BeamSpotProducer/plugins/BeamSpotOnlineProducer.cc

```diff
@@ -129,25 +161,59 @@ bool BeamSpotOnlineProducer::shouldShout(const edm::Event& iEvent) const {
   return true;  // Default to "shout" if the record is missing
 }
 
-bool BeamSpotOnlineProducer::processTransientRecord(const edm::EventSetup& iSetup,
-                                                    reco::BeamSpot& result,
-                                                    bool shoutMODE) {
-  auto const& spotDB = iSetup.getData(beamTransientToken_);
+bool BeamSpotOnlineProducer::processRecords(const edm::Event& iEvent,
+                                            const edm::EventSetup& iSetup,
+                                            reco::BeamSpot& result,
+                                            bool shoutMODE) {
+  auto const spotDBLegacy = getBeamSpotFromRecord(beamTokenLegacy_, iEvent, iSetup);
+  auto const spotDBHLT = getBeamSpotFromRecord(beamTokenHLT_, iEvent, iSetup);
+  auto const& spotDB = chooseBS(spotDBLegacy, spotDBHLT);
 
-  if (spotDB.beamType() != 2) {
-    if (shoutMODE && beamTransientRcdESWatcher_.check(iSetup)) {
+  if (spotDB.beamType() != reco::BeamSpot::Tracker) {
+    if (shoutMODE && (beamLegacyRcdESWatcher_.check(iSetup) || beamHLTRcdESWatcher_.check(iSetup))) {
       edm::LogWarning("BeamSpotOnlineProducer")
           << "Online Beam Spot producer falls back to DB value due to fake beamspot in transient record.";
     }
     return true;  // Trigger fallback to DB
   }
 
   // Create BeamSpot from transient record
-  createBeamSpotFromTransient(spotDB, result);
+  createBeamSpotFromRecord(spotDB, result);
   return false;  // No fallback needed
 }
 
-void BeamSpotOnlineProducer::createBeamSpotFromTransient(const BeamSpotObjects& spotDB, reco::BeamSpot& result) const {
+template <typename TokenType>
+const BeamSpotOnlineObjects BeamSpotOnlineProducer::getBeamSpotFromRecord(const TokenType& token,
+                                                                          const Event& event,
+                                                                          const EventSetup& setup) {
+  const auto& bs = setup.getData(token);
+  if (bs.sigmaZ() < sigmaZThreshold_ || bs.beamWidthX() < sigmaXYThreshold_ || bs.beamWidthY() < sigmaXYThreshold_ ||
+      bs.beamType() != reco::BeamSpot::Tracker) {
+    return fakeBS_;
```

Comment:
in the `OnlineBeamSpotESProducer` we instrumented all the cases in which a fake BS is returned because of https://github.com/cms-sw/cmssw/issues/45555, e.g. here:

https://github.com/cms-sw/cmssw/blob/b8776f576fdc5aa8e36d0396fd3aa567ea05877f/RecoVertex/BeamSpotProducer/plugins/OnlineBeamSpotESProducer.cc#L166-L168

or here

https://github.com/cms-sw/cmssw/blob/b8776f576fdc5aa8e36d0396fd3aa567ea05877f/RecoVertex/BeamSpotProducer/plugins/OnlineBeamSpotESProducer.cc#L148-L149

please do the same.

Reply:
I added back the warnings when the record does not pass the sanity check or the time threshold, I do not know however how to replicate ```tryToGetRecord```